### ISSUE TITLE
baseline estimation with QuantRegression

### ISSUE THREAD

**Issue by njbernstein** [statsmodels/statsmodels#4341]
I'm trying to implement the following 
![image](https://user-images.githubusercontent.com/6864886/37435564-0d1c0998-27a1-11e8-81fd-e127d2dc5456.png) from [this paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3064906/) in python. 

```
x = some_list #12000 measurements or so
place_holder = range(0,len(x))
data = {'x' : x, 'place_holder' : place_holder}
df = pd.DataFrame(data=data)
mod = smf.quantreg(formula, df)
res = mod.fit(q=quantile)
```

I run it with the following formulas and get the following aic values out:
```
formula: x ~ place_holder 
aic: 394623.170417
formula: x ~ place_holder  + I(place_holder ** 2)
aic: 394427.966987

```
and then get this error:
```
formula: x ~ place_holder  + I(place_holder ** 2) + I(place_holder ** 3)

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-118-cf144633801a> in <module>()
     81 
     82     return output
---> 83 c = quantile_reg_baseline_estimation(a.sample_data, 20, .1)
     84 #d = quantile_reg_baseline_estimation(a.sample_data - c, 20, .1)

<ipython-input-118-cf144633801a> in quantile_reg_baseline_estimation(x, possible_degrees_of_freedom, quantile)
     66         mod = smf.quantreg(formula, df)
     67         #mod = smf.quantreg('dens ~ temp + I(temp ** 2.0)', df)
---> 68         res = mod.fit(q=quantile)
     69         models.append(res)
     70         res.llf = loglike(res, res.params, mod.endog, mod.exog, scale=None)

/locus/home/nbernstein/main/locus-pipe/ve/lib/python2.7/site-packages/statsmodels/regression/quantile_regression.pyc in fit(self, q, vcov, kernel, bandwidth, max_iter, p_tol, **kwargs)
    177             resid = np.abs(resid)
    178             xstar = exog / resid[:, np.newaxis]
--> 179             diff = np.max(np.abs(beta - beta0))
    180             history['params'].append(beta)
    181             history['mse'].append(np.mean(resid*resid))

ValueError: operands could not be broadcast together with shapes (4,) (3,) 
    
```
When I try formulas for cubics and above.

I'm sure this error is due to the rank of my matrix being less than the actual shape of my matrix, but this is expected. I'm wondering if a pull request to change the starting beta to the shape of the input matrix would be reasonable, or how I might fix my issue otherwise?

**Comment by josef-pkt** [comment]
I guess this is #2597, i.e. exog is singular and matrix rank is smaller than number of columns.

One thing you could check is whether the scaling of `place_holder` is so that third power is collinear with lower powers. In that case, rescaling `place_holder` most likely helps.

AFAICS, there is no option to provide start_params to `fit`

**Comment by njbernstein** [comment]
What sort of rescaling would you suggest using in that case?

**Comment by josef-pkt** [comment]
numpy polynomial classes default to rescaling to an interval [-1, 1]
Depending on the values (meaning of the variable) and the polynomial order [0, 2] or [0, 5] or standardizing to mean 0 and variance 1 should work.

What should be avoided are large values (in absolute terms) that where polynomials become very large, e.g. 100*3 is large compared to 1 and 100. 
(I often check that plotting np.vander(x, order) is not be dominated by one curve)

**Comment by njbernstein** [comment]
I tried using a zscore based on this https://stats.stackexchange.com/questions/35071/what-is-rank-deficiency-and-how-to-deal-with-it

Led to the same issue though

**Comment by josef-pkt** [comment]
can you post the data to see what's going on?

you could change the line in `fit` from
`beta = np.ones(exog_rank)`
to 
`beta = np.ones(exog.shape[1])`
to see what the result is in that case.

However, that would only remove one symptom and still leave the underlying rank deficiency problem.

**Comment by njbernstein** [comment]
I had done that previously and it works.

I figured out my issue. The baseline estimation is working fine, but I have some significant troughs in my data that it is treats as the baseline at points. we are all good here thanks for your help. @josef-pkt really appreciate it

**Comment by josef-pkt** [comment]
just another general recommendation, or something to try: use splines instead of polynomials if there are larger fluctuations.

I got good results with patsy splines in the cases that I tried. We don't have penalization yet, but choosing knots at every k'th observations works well in many cases (with not too many knots).
The main disadvantage is that it doesn't allow extrapolation, i.e. prediction outside of the fit data.
